blockchain/state_manager/src/utils.rs

```diff
@@ -100,3 +109,270 @@ where
+
+    let mut sset: Vec<ChainSectorInfo> = Vec::new();
+    let for_each = |i: u64, sector_chain: &miner::SectorOnChainInfo| -> Result<(), String> {
+        if let Some(ref mut s) = filter {
+            let i = i
+                .try_into()
+                .map_err(|_| "Could not convert from index to usize")?;
+            if s.get(i) {
+                return Ok(());
+            }
+        }
+        sset.push(ChainSectorInfo {
+            info: sector_chain.info.to_owned(),
+            id: i,
+        });
+        Ok(())
+    };
+    amt.for_each(for_each)
+        .map_err(|err| Error::Other(format!("Error Processing ForEach {:}", err)))?;
+
+    Ok(sset)
+}
+
+pub fn miner_sector_info<DB>(
+    state_manager: &StateManager<DB>,
+    address: &Address,
+    sector_number: &SectorNumber,
+    tipset: &Tipset,
+) -> Result<Option<SectorOnChainInfo>, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get miner sector info) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    miner_actor_state
+        .get_sector(state_manager.get_block_store_ref(), *sector_number)
+        .map_err(|err| Error::State(format!("(get sset) failed to get actor state: {:}", err)))
+}
+
+pub fn precommit_info<DB>(
+    state_manager: &StateManager<DB>,
+    address: &Address,
+    sector_number: &SectorNumber,
+    tipset: &Tipset,
+) -> Result<SectorPreCommitOnChainInfo, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get precommit info) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    let precommit_info = miner_actor_state
+        .get_precommitted_sector(state_manager.get_block_store_ref(), *sector_number)
+        .map_err(|err| {
+            Error::Other(format!(
+                "(precommit info) failed to load miner actor state: %{:}",
+                err
+            ))
+        })?;
+    Ok(precommit_info.ok_or_else(|| Error::Other("precommit not found".to_string()))?)
+}
+
+pub fn get_miner_info<DB>(
+    state_manager: &StateManager<DB>,
+    tipset: &Tipset,
+    address: &Address,
+) -> Result<MinerInfo, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get miner info) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    Ok(miner_actor_state.info)
+}
+
+pub fn get_miner_deadlines<DB>(
```

Comment:
where is this function coming from? I'm having a hard time mapping some of these functions in this file

Reply:
https://github.com/filecoin-project/lotus/blob/c459a13dcac66941c61b0c67105468e83e1bf873/chain/stmgr/utils.go#L305